### PR TITLE
added typebox and basic user apui

### PR DESCRIPTION
**Pull request by mundanecoder**
added typeBox not zod ,
created usermodel and basic createuser api

### CODE REVIEW

File: src/Models/UserModel.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    lastName: {
+      type: String,
+      required: true,
+    },
+    fullName: {
+      type: String,
+      required: true,
+    },
+    phoneNumbers: {
```
Comment (mridul1024) [inline src/Models/UserModel.ts:38]: I think this is a typo, but if it's not, convert the type of **"phoneNumbers"** to a string array if you are planning to store multiple phone numbers for a particular user.

File: src/Models/UserModel.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      type: Array,
+    },
+    age: {
+      type: Number,
+    },
+    gender: {
+      type: String,
+    },
+    email: {
```
Comment (mridul1024) [inline src/Models/UserModel.ts:49]: Add a validation logic for this field in order evaluate the correctness of the email pattern. You check out the validation property from the mongoose docs. That will allow you to include a check as well as a custom validation message for this field.

File: src/Models/UserModel.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    userDeleted: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  {
+    timestamps: true,
+  }
+);
+
```
Comment (mridul1024) [inline src/Models/UserModel.ts:70]: Add a **Mongoose Virtual Property** which can be computed by combining other property values. Create a virtual property **_fullname_** which will be created by the existing **firstname** and **lastname** properties.

File: src/Models/UserModel.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+export interface IUserDocument extends IUser, Document {
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+const UserSchema = new Schema<IUserDocument>(
+  {
+    clerkId: {
+      type: "string",
```
Comment (mridul1024) [inline src/Models/UserModel.ts:23]: Is there something wrong with this type?

File: src/Controller/user/saveUserData.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        const userExist = await UserDB.findOne({
+          email: userData?.emailAddresses[0].emailAddress,
+        });
+
+        if (!userExist) {
+          let newUser = new UserDB({
+            firstname: userData?.firstName,
+            lastName: userData?.lastName,
+            fullName: userData?.fullName,
+            phoneNumbers: userData?.phoneNumbers ? userData?.phoneNumbers : "",
```
Comment (mridul1024) [inline src/Controller/user/saveUserData.ts]: Nullish coalescing operator (??) use karo...

File: src/Controller/user/saveUserData.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        },
+      },
+    },
+    async (req: FastifyRequest, rep: FastifyReply) => {
+      const userData = req.user;
+
+      console.log(userData?.emailAddresses);
+
+      try {
+        const userExist = await UserDB.findOne({
```
Comment (mridul1024) [inline src/Controller/user/saveUserData.ts]: Create a **Services** folder and a **Repository** folder. Follow the same structure as your Controller space. Move the logical block to services and the DB accessor functions to repositories. The Controller should be thin, and the services should handle the logical part.

**Review by mridul1024 — CHANGES_REQUESTED** [review]
Comments add kar dia hei @mundanecoder , thorha sa alterations kar do, fir merge karte hei. And wrong branch pe pull request hei, close kar k dev-staging se branch create karo and push uspe, fir pull request from your branch to dev-staging.

File: src/Controller/user/saveUserData.ts (hunk 6 above)
Comment (mundanecoder) [reply src/Controller/user/saveUserData.ts]: sure i will do it right away !!

File: src/Models/UserModel.ts (hunk 3 above)
Comment (mundanecoder) [reply src/Models/UserModel.ts:70]: if the userdata already has a fullname field , do i need the virtual property ?

File: src/Models/UserModel.ts (hunk 4 above)
Comment (mundanecoder) [reply src/Models/UserModel.ts:23]: i am not sure, i kept in order to be out of any problem if we need it for some referencing in future.

**Pull request closed without merging**